target.go

```diff
@@ -186,25 +225,50 @@ func (s *targetServer) obliviousQueryHandler(w http.ResponseWriter, r *http.Requ
 		return
 	}
 
-	packedResponse, err := s.resolveQuery(query)
+	queryParseAndDecryptionCompleteTime := time.Now().UnixNano()
+	timestamp.TargetQueryDecryptionTime = queryParseAndDecryptionCompleteTime
+
+	resolverChosen := s.resolver[chosenResolver]
+	packedResponse, err := s.resolveQueryUsingNS(query, resolverChosen)
 	if err != nil {
 		log.Println("Failed resolving DNS query:", err)
 		http.Error(w, http.StatusText(http.StatusInternalServerError), http.StatusInternalServerError)
```

Comment:
Should we also log an error here?

Reply:
We do have a log statement before the error is returned which show up on the GAE logs